app/src/test/kotlin/main/MainViewModelTest.kt

```diff
@@ -0,0 +1,129 @@
+package main
+
+import androidx.paging.ExperimentalPagingApi
+import app.cash.turbine.test
+import com.architectcoders.arquitectomarvel.ui.main.MainViewModel
+import com.architectcoders.arquitectomarvel.ui.main.pagination.CharacterRemoteMediator
+import com.architectcoders.usecases.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import kotlin.time.ExperimentalTime
+
+@RunWith(MockitoJUnitRunner::class)
+class MainViewModelTest {
+
+    @ExperimentalPagingApi
+    @Mock
+    private lateinit var characterRemoteMediator: CharacterRemoteMediator
+
+    @Mock
+    private lateinit var getPagingSourceFromCharacterEntity: GetPagingSourceFromCharacterEntity
+
+    @Mock
+    private lateinit var isPasswordAlreadyStored: IsPasswordAlreadyStored
+
+    @Mock
+    private lateinit var saveCredentials: SaveCredentials
+
+    @Mock
+    private lateinit var deleteCredentials: DeleteCredentials
+
+    @Mock
+    private lateinit var isPasswordCorrect: IsPasswordCorrect
+
+    @Mock
+    private lateinit var isRecoveryHintCorrect: IsRecoveryHintCorrect
+
+    @Mock
+    private lateinit var deleteAllLocalFavoriteCharacter: DeleteAllLocalFavoriteCharacter
+
+    @Mock
+    private lateinit var listener: (Boolean) -> Unit
+
+    private lateinit var mainViewModel: MainViewModel
+
+    @ExperimentalPagingApi
+    @Before
+    fun setup() {
+        mainViewModel = MainViewModel(
+            Dispatchers.Unconfined,
+            characterRemoteMediator,
+            getPagingSourceFromCharacterEntity,
+            isPasswordAlreadyStored,
+            saveCredentials,
+            deleteCredentials,
+            isPasswordCorrect,
+            isRecoveryHintCorrect,
+            deleteAllLocalFavoriteCharacter
+        )
+    }
+
+    @ExperimentalTime
+    @Test
+    fun `confirm if the first passwordState state is the INITIAL_STATE value`(): Unit =
+        runBlocking {
+            mainViewModel.passwordState.test {
+                assertEquals(awaitItem(), MainViewModel.PasswordState.INITIAL_STATE)
+                cancelAndConsumeRemainingEvents()
+            }
+        }
+
+    @ExperimentalTime
+    @Test
+    fun `confirm if the passwordState state has been updated`(): Unit = runBlocking {
+        mainViewModel.updatePasswordState(MainViewModel.PasswordState.SUCCESSFUL)
+        mainViewModel.passwordState.test {
+            assertEquals(awaitItem(), MainViewModel.PasswordState.SUCCESSFUL)
+            cancelAndConsumeRemainingEvents()
+        }
+    }
+
+    @Test
+    fun `confirm if password is already stored`(): Unit = runBlocking {
```

Comment:
In my opinion is more readable so:
```
    @Test
    fun `confirm if password is already stored 2`() = runBlocking {
        whenever(isPasswordAlreadyStored.invoke(Unit)).thenReturn(true)
        mainViewModel.ifDeviceNeitherHaveBiometricLoginNorPassword {
            launch {
                assert(it)
            }
        }
    }
```

Reply:
Thanks @joseantoniocampillo I think that is another way to do it :smile: